apps/aecore/lib/aecore/sync/chain.ex

```diff
@@ -0,0 +1,141 @@
+defmodule Aecore.Sync.Chain do
+  @moduledoc """
+  Implements all the functions regarding the Chain structure of the SyncTask
+  """
+
+  alias Aecore.Chain.Header
+  alias Aecore.Chain.BlockValidation
+  alias Aecore.Sync.Task
+  alias __MODULE__
+
+  @type peer_id :: pid()
+  @type chain_id :: reference()
+  @type height :: non_neg_integer()
+  @type hash :: binary()
+  @type chain :: %{height: height(), hash: hash()}
+
+  @type t :: %Chain{chain_id: chain_id(), peers: list(), chain: list(chain())}
+
+  defstruct chain_id: nil,
+            peers: [],
+            chain: []
+
+  @spec init_chain(peer_id(), Header.t()) :: t()
+  def init_chain(peer_id, header) do
+    init_chain(Kernel.make_ref(), [peer_id], header)
```

Comment:
is Kernel added not to confuse these functions for local ones?

Reply:
Yes